blog/executor-plugins.md

```diff
@@ -0,0 +1,348 @@
+
+Add new build entry under `.goreleaser.yaml`:
+
+```yaml
+builds:
+  - id: gh
+    main: cmd/gh/main.go
+    binary: executor_gh_{{ .Os }}_{{ .Arch }}
+
+    no_unique_dist_dir: true
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
+    goarch:
+      - amd64
+      - arm64
+    goarm:
+      - 7
+```
+
+Now, we need to distribute our plugins. As we mentioned earlier, a plugin repository can be any static file server. The [kubeshop/botkube-plugins-template](https://github.com/kubeshop/botkube-plugins-template) repository comes with two [GitHub Actions](https://github.com/features/actions):
+
+1. The [`.github/workflows/release.yml`](https://github.com/kubeshop/botkube-plugins-template/blob/main/.github/workflows/release.yml) action, which builds the plugin binaries and index file each time a new tag is pushed.
+2. The [`.github/workflows/pages-release.yml`](https://github.com/kubeshop/botkube-plugins-template/blob/main/.github/workflows/pages-release.yml) action, which updates GitHub Pages with plugin binaries and index file each time a new tag is pushed.
+
+To cut a new release, you need to commit all your work and tag a new commit:
+
+```bash
+git add -A
+git commit -m "Implement gh executor"
+git tag v1.0.0
+```
+
+Next, let's push our changes and the new tag:
+
+```bash
+git push
+git push --tags
+```
+
+This triggers GitHub Action:
+
+![GitHub action that creates a new release](./assets/release-job.png)
+
+**What this automation does under the hood**
+
+This automation:
+
+1. Installs the latest [GoReleaser](https://goreleaser.com/) tool.
+2. Builds all plugin binaries defined in the `.goreleaser.yaml` file.
+3. Generates an index file using the [Botkube helper tool](https://docs.botkube.io/plugin/repo#generate-index-file).
+4. Generates a release description.
+5. Uses the [`gh`](https://cli.github.com) CLI to create a new GitHub release.
+
+### Use the `gh` executor
+
+In the description of a new GitHub release, you will see the repository URL that you can use within Botkube.
+
+#### Steps
+
+1. Follow one of our [installation guides](https://docs.botkube.io/installation). Once you reach the Botkube deployment step, add flags specified in the steps below to `helm install`.
+2. Export required environment variables:
+
+   :::info
+   Follow the official GitHub guide on how to create a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). To be able to create GitHub issues, add the `repo` permission.
+   :::
+
+   ```bash
+   export REPOSITORY={repo} # format OWNER/REPO_NAME, e.g. kubeshop/botkube
+   export GITHUB_TOKEN={token}
+   export PLUGINS_URL={plugin_index_url}
+   ```
+
+3. Add the `gh` executor related configuration:
+
+   ```bash
+   # example configuration for `gh` executor
+   -f https://gist.github.com/mszostok/88e6852f93429928d1183e52226921fc/raw \
```

Comment:
We can get rid of `-f https://gist.github.com/mszostok/88e6852f93429928d1183e52226921fc/raw` if we will embed the issue template. It will make our code longer, though.

Reply:
Yes, I think for visibility it'd be better to remove the gist 👍